.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+          gh release create ${{ needs.extract-version-details.outputs.aliased-version }} \
+            --notes "The Captain ${{ needs.extract-version-details.outputs.aliased-version }} release and tag exist to provide an easy way to download the latest ${{ needs.extract-version-details.outputs.aliased-version }}.x.x release of Captain. For example, you can always download the latest Linux x86 ${{ needs.extract-version-details.outputs.aliased-version }} release at this URL: https://github.com/rwx-research/captain/releases/download/${{ needs.extract-version-details.outputs.aliased-version }}/captain-linux-x86_64. (Look at the assets attached to this release to find the other available downloads.) This release and its assets are updated whenever a new ${{ needs.extract-version-details.outputs.aliased-version }}.x.x version of Captain is released." \
+            --title "Captain ${{ needs.extract-version-details.outputs.aliased-version }}"
+      - name: Push the assets from the full version release to the aliased release
+        if: ${{ needs.extract-version-details.outputs.aliased-version != '' }}
+        run: |
+          gh release download ${{ needs.extract-version-details.outputs.full-version }}
+          gh release upload ${{ needs.extract-version-details.outputs.aliased-version }} captain-* --clobber
+
+  update-cloudfront:
+    runs-on: ubuntu-latest
+    environment: Production
+    if: ${{ always() && needs.extract-version-details.result == 'success' && needs.publish-release.result == 'success' }}
+    needs:
+      - extract-version-details
+      - publish-release
+    permissions:
+      id-token: read
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::152560469324:role/captain-cli-cd
+          role-duration-seconds: 900
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - name: Download the binaries
+        run: gh release download ${{ needs.extract-version-details.outputs.full-version }}
+      - name: Upload the binaries to S3 and invalidate cloudfront
+        run: |
+          updatedPaths=""
+
+          for binary in captain-*
+          do
+            cleanedBinary=${binary//.exe/}
+            os=$(echo "$cleanedBinary" | cut -d '-' -f 2)
+            arch=$(echo "$cleanedBinary" | cut -d '-' -f 3)
+            version="${{ needs.extract-version-details.outputs.full-version }}"
+
+            if [[ "$arch" == "x86_64" ]]; then
+              oldArch="amd64"
+            elif [[ "$arch" == "aarch64" ]]; then
+              oldArch="arm64"
+            else
+              echo "unmapped arch: $arch"
+              exit 1
+            fi
+
+            if [[ "$os" == "windows" ]]; then
+              extension=".exe"
+            fi
+
+            oldPath="captain-$os-$oldArch-$version"
```

Comment:
💭 might be worth adding a comment indicating the extension is intentionally missing